pkg/workceptor/kubernetes.go

```diff
@@ -533,15 +533,32 @@ mainLoop:
 
 					return
 				case containerState.Terminated != nil:
-					// We got EOF and the pod terminated, we will log the terminated information
-					if containerState.Terminated.ExitCode != 0 {
+					reason := containerState.Terminated.Reason
+
+					// Whitelist: "Completed" and "Error" mean the program ran to completion
+					// Everything else (OOMKilled, Evicted, etc.) means execution was interrupted
+					if reason != "Completed" && reason != "Error" {
 						kw.GetWorkceptor().nc.GetLogger().Info("%s/%s: %s has terminated, with nonzero exit code: %v, terminated reason: %v and terminated message: %v",
```

Comment:
This line will log that the container terminated with a `nonzero exit code`. But we're no longer checking the exit code value, so I don't think it makes sense to assume it's nonzero and hard-code that in the log message. (In practice it's _probably_ going to be nonzero if the reason wasn't `Completed` but it's also an assumption)

I think this log message would work better to log that the container terminated with an `unexpected reason` since that's what we're checking. That said, we have another `GetLogger.Info()` just after this `if {}` block (`// Log completion`) with most of the same information (it's just omitting the `.Message`). 

Can we consolidate these into one `GetLogger.Info()` on completion?